imgui-SFML.cpp

```diff
@@ -688,39 +688,61 @@ void Image(const sf::RenderTexture& texture, const sf::Vector2f& size, const sf:
 /////////////// Image Overloads for sf::Sprite
 
 void Image(const sf::Sprite& sprite, const sf::Color& tintColor, const sf::Color& borderColor) {
-	Image(sprite, sf::Transform::Identity, tintColor, borderColor);
+	Image(sprite, sprite.getGlobalBounds().getSize(), sf::Transform::Identity, tintColor, borderColor);
 }
 
-void Image(const sf::Sprite& sprite, const sf::Transform& additionalTransform, const sf::Color& tintColor,
-    const sf::Color& borderColor)
+void Image(const sf::Sprite& sprite, const sf::Vector2f& size, const sf::Color& tintColor,
+    const sf::Color& borderColor) {
+	Image(sprite, size, sf::Transform::Identity, tintColor, borderColor);
+}
+
+void Image(const sf::Sprite& sprite, const sf::Vector2f& size, const sf::Transform& additionalTransform,
+    const sf::Color& tintColor, const sf::Color& borderColor)
 {
-    const sf::Texture* texturePtr = sprite.getTexture();
-    // sprite without texture cannot be drawn
-    if (!texturePtr) {
+    const sf::IntRect& textureRect = sprite.getTextureRect();
+
+    // sprite without or empty texture cannot be drawn
+    if (!sprite.getTexture() ||
+        textureRect.width <= 0 || textureRect.height <= 0 ||
+        size.x <= 0 || size.y <= 0) {
         return;
     }
 
+    // simply taken form ImGui::Image
     ImGuiWindow* window = GetCurrentWindow();
     if (window->SkipItems)
         return;
 
-    const sf::Texture& texture = *texturePtr;
-    ImTextureID textureID = convertGLTextureHandleToImTextureID(texture.getNativeHandle());
-    sf::Vector2f textureSize = static_cast<sf::Vector2f>(texture.getSize());
-    const sf::IntRect& textureRect = sprite.getTextureRect();
+    ImRect itemBB(window->DC.CursorPos.x, window->DC.CursorPos.y,
+        window->DC.CursorPos.x + size.x, window->DC.CursorPos.y + size.y);
+    if (borderColor.a > 0)
+    {
+        itemBB.Max.x += 2;
+        itemBB.Max.y += 2;
+    }
+
+    ItemSize(itemBB);
+    if (!ItemAdd(itemBB, 0))
+        return;
+
+    // prepare uv coordinates
+    const sf::Vector2f textureSize = static_cast<sf::Vector2f>(sprite.getTexture()->getSize());
```

Comment:
Texture can be NULL here. Please add a check and just "return" if it's the case

Reply:
It is already checked in the first condition of the function...